File: app/src/main/java/com/example/despensacx/FormatUtils.java
package com.example.despensacx;

import java.text.NumberFormat;
import java.util.Locale;

public class FormatUtils {

    private static final NumberFormat currencyFormatter;

    static {
        // Formato para México / Español ($4,000.00)
        currencyFormatter = NumberFormat.getCurrencyInstance(new Locale("es", "MX"));
    }

    public static String formatCurrency(double amount) {
        synchronized (currencyFormatter) {
            return currencyFormatter.format(amount);
        }
    }
}

File: app/src/main/kotlin/com/example/despensacx/utils/FormatUtils.kt
package com.example.despensacx.utils

import java.text.NumberFormat
import java.util.Locale

object FormatUtils {
    private val currencyFormatter: NumberFormat = NumberFormat.getCurrencyInstance(Locale("es", "MX"))

    @JvmStatic
    fun formatCurrency(amount: Double): String {
        synchronized(currencyFormatter) {
            return currencyFormatter.format(amount)
        }
    }
}
